app/src/main/kotlin/com/wire/android/ui/common/topappbar/AppTopBarWithSearchBarLayout.kt

```diff
@@ -0,0 +1,269 @@
+    onInputClicked: () -> Unit,
+    onCloseSearchClicked: () -> Unit,
+    appTopBar: @Composable () -> Unit,
+) {
+    Surface(
+        modifier = Modifier
+            .fillMaxWidth()
+            .wrapContentHeight(),
+        shadowElevation = if (isTopBarVisible) dimensions().spacing8x else 0.dp
+    ) {
+        Column(
+            Modifier
+                .wrapContentSize()
+                .animateContentSize(animationSpec = snap())
+        ) {
+            AnimatedVisibility(
+                visible = isTopBarVisible,
+                enter = expandVertically(),
+                exit = shrinkVertically(),
+            ) {
+                Box(
+                    Modifier
+                        .wrapContentSize()
+                ) {
+                    appTopBar()
+                }
+            }
+
+            AnimatedVisibility(!isSearchBarCollapsed) {
+                Box(
+                    modifier = Modifier
+                        .wrapContentHeight()
+                        .fillMaxWidth()
+                        .background(MaterialTheme.wireColorScheme.background)
+                ) {
+                    val interactionSource = remember {
+                        MutableInteractionSource()
+                    }
+
+                    val focusManager = LocalFocusManager.current
+
+                    SearchBarInput(
+                        placeholderText = searchBarHint,
+                        text = searchQuery,
+                        onTextTyped = onSearchQueryChanged,
+                        leadingIcon = {
+                            AnimatedContent(isTopBarVisible) { isVisible ->
+                                IconButton(onClick = {
+                                    if (isVisible) {
+                                        focusManager.clearFocus()
+
+                                        onCloseSearchClicked()
+                                    }
+                                }) {
+                                    Icon(
+                                        painter = if (isVisible) painterResource(id = R.drawable.ic_search_icon) else painterResource(id = R.drawable.ic_arrow_left),
+                                        contentDescription = stringResource(R.string.content_description_conversation_search_icon),
+                                        tint = MaterialTheme.wireColorScheme.onBackground
+                                    )
+                                }
+                            }
+                        },
+                        placeholderTextStyle = if (isTopBarVisible) LocalTextStyle.current.copy(textAlign = TextAlign.Center) else LocalTextStyle.current.copy(
+                            textAlign = TextAlign.Start
+                        ),
+                        textStyle = if (isTopBarVisible) LocalTextStyle.current.copy(textAlign = TextAlign.Center) else LocalTextStyle.current.copy(
+                            textAlign = TextAlign.Start
```

Comment:
Maybe extract the textalign to a variable and apply to both the same text align, placeholder and text